### PR TITLE
Bug and vulnerability fixes

### PR DESCRIPTION
**Pull request by erademacher**
Two commits, one of which fixes a series of dependabot vulnerabilities by updating the `net` library, and one which addresses issue #80.

- - -
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/85)

### CODE REVIEW

File: internal/provider/cluster_resource.go [hunk 1]
```diff
@@ -331,6 +331,14 @@ func (r *clusterResource) Read(ctx context.Context, req resource.ReadRequest, re
 		return
 	}
 	clusterID := cluster.ID.ValueString()
+	// In case this was an import, validate the ID format.
+	if !uuidRegex.MatchString(clusterID) {
+		resp.Diagnostics.AddError(
+			"Unexpected cluster ID format",
+			fmt.Sprintf("'%s' is not a invalid cluster ID format. Expected UUID.", clusterID),
```
Comment (jenngeorge) [inline internal/provider/cluster_resource.go:338]: typo:  "invalid" should be "valid" for "is not a valid cluster ID format."

File: internal/provider/cockroach_cluster_data_source.go [hunk 2]
```diff
@@ -157,18 +157,27 @@ func (d *clusterDataSource) Read(ctx context.Context, req datasource.ReadRequest
 		)
 		return
 	}
+	clusterID := cluster.ID.ValueString()
+	if !uuidRegex.MatchString(clusterID) {
+		resp.Diagnostics.AddError(
+			"Unexpected cluster ID format",
+			fmt.Sprintf("'%s' is not a invalid cluster ID format. Expected UUID.", clusterID),
```
Comment (jenngeorge) [inline internal/provider/cockroach_cluster_data_source.go:164]: typo: "invalid" -> "valid"

**Review by jenngeorge — COMMENTED** [review]
I just merged the new database resource -- could you also update the cluster IDs for https://github.com/cockroachdb/terraform-provider-cockroach/blob/56eb6be6764bac350b5ac802abcf69f74c6285ce/internal/provider/database_resource_test.go#L51-L63 & update `uuidRegex` -> `uuidRegexString` for https://github.com/cockroachdb/terraform-provider-cockroach/blob/56eb6be6764bac350b5ac802abcf69f74c6285ce/internal/provider/database_resource.go#L43 as part of the `Validate cluster IDs` commit?

**Review by erademacher — COMMENTED** [review]
Oops! Thanks for catching that. Fixed.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/85#-:-NPyQ_2f96lxdAkRl1Ko:b-gl0arf)* status: 0 of 16 files reviewed, all discussions resolved

**Review by jenngeorge — APPROVED** [review]
:lgtm: !

> Reviewed 16 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/85#-:-NPyUOJODikNb9XEIpKM:b593w5u)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @erademacher)

**Review by erademacher — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/terraform-provider-cockroach/85#-:-NPyVi-E4zrUSL-qrnip:b-814g8k)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @erademacher)

**Pull request merged**